src/transformers/testing_utils.py

```diff
@@ -3482,11 +3488,17 @@ def _get_test_info():
         tb_next = tb
     test_traceback = tb
 
+    origin_method_being_patched = frame_of_patched_obj.frame.f_locals["orig_method"]
+
+    # An iterable of type `traceback.StackSummary` with each element of type `FrameSummary`
     stack = traceback.extract_stack()
+    # The frame which calls `the original method being patched`
+    caller_frame = None
+    # From the most inner (i.e. recent) frame to the most outer frame
+    for frame in reversed(stack):
+        if origin_method_being_patched.__name__ in frame.line:
+            caller_frame = frame
 
-    # The frame that calls this patched method (it may not be the test method)
-    # -1: `_get_test_info`; -2: `patched_xxx`; -3: the caller to `patched_xxx`
-    caller_frame = stack[-3]
```

Comment:
This was a bit hardcoded, and it turns out that this causes issue when `@require_read_token` is applied to a test class, where we will get

stack[-1]: `_get_test_info`
stack[-2]: `patched`
stack[-3]: `wrapper` (from `def require_read_token`)
stack[-4]: the caller to the method that is wrapped (in this case, it's `self.assertEqual`)